### PR TITLE
Split QC steps into different parts (picard, samtools, multiQC stats …

### PR DESCRIPTION
**Pull request by jonca79**
…gethering) in preparation for WP3 integration.

### CODE REVIEW

File: src/Snakemake/rules/QC/picard.smk [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        bam="DNA_bam/{sample}-ready.bam",
+        intervals=config["bed"]["intervals"],
+    output:
+        "qc/{sample}/{sample}.HsMetrics.txt",
+    log:
+        "logs/qc/picard/HsMetrics/{sample}.log",
+    singularity:
+        config["singularity"]["picard"]
+    shell:
+        "(java -Xmx4g -jar /opt/conda/share/picard-2.20.1-0/picard.jar CollectHsMetrics BAIT_INTERVALS={input.intervals} TARGET_INTERVALS={input.intervals} INPUT={input.bam} OUTPUT={output}) &> {log}"
```
Comment (Smeds) [inline src/Snakemake/rules/QC/picard.smk:16]: [Wrapper 0.72.0/bio/picard/collecthsmetrics](https://snakemake-wrappers.readthedocs.io/en/0.72.0/wrappers/picard/collecthsmetrics.html) ?

File: src/Snakemake/rules/QC/picard.smk [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        bam="DNA_bam/{sample}-ready.bam",
+    output:
+        txt="qc/{sample}/{sample}.insert_size_metrics.txt",
+        pdf="qc/{sample}/{sample}.insert_size_histogram.pdf",
+    log:
+        "logs/qc/picard/InsertSize/{sample}.log",
+    singularity:
+        config["singularity"]["picard"]
+    shell:
+        "(java -Xmx4g -jar /opt/conda/share/picard-2.20.1-0/picard.jar CollectInsertSizeMetrics INPUT={input.bam} O={output.txt} H={output.pdf}) &> {log}"
```
Comment (Smeds) [inline src/Snakemake/rules/QC/picard.smk]: [Wrapper 0.72.0/bio/picard/collectinsertsizemetrics](https://snakemake-wrappers.readthedocs.io/en/0.72.0/wrappers/picard/collectinsertsizemetrics.html)

File: src/Snakemake/rules/QC/picard.smk [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        bam="DNA_bam/{sample}-ready.bam",
+        ref=config["reference"]["ref"],
+    output:
+        "qc/{sample}/{sample}.alignment_summary_metrics.txt",
+    log:
+        "logs/qc/picard/AlignmentSummaryMetrics/{sample}.log",
+    singularity:
+        config["singularity"]["picard"]
+    shell:
+        "(java -Xmx4g -jar /opt/conda/share/picard-2.20.1-0/picard.jar CollectAlignmentSummaryMetrics INPUT={input.bam} R={input.ref} OUTPUT={output}) &> {log}"
```
Comment (Smeds) [inline src/Snakemake/rules/QC/picard.smk]: [Wrapper 0.72.0/bio/picard/collectalignmentsummarymetrics](https://snakemake-wrappers.readthedocs.io/en/0.72.0/wrappers/picard/collectalignmentsummarymetrics.html)?

File: src/Snakemake/rules/QC/picard.smk [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    output:
+        summary="qc/{sample}/{sample}.gc_bias.summary_metrics.txt",
+        gc="qc/{sample}/{sample}.gc_bias.metrics.txt",
+        pdf="qc/{sample}/{sample}.gc_bias.metrics.pdf",
+    log:
+        "logs/qc/picard/GcBiasSummaryMetrics/{sample}.log",
+    singularity:
+        config["singularity"]["picard"]
+    shell:
+        "(java -Xmx4g -jar /opt/conda/share/picard-2.20.1-0/picard.jar CollectGcBiasMetrics I={input.bam} R={input.ref} O={output.gc} CHART={output.pdf} S={output.summary}) &> {log}"
```
Comment (Smeds) [inline src/Snakemake/rules/QC/picard.smk:98]: Could not find a existing wrapper

File: src/Snakemake/rules/QC/samtools.smk [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    output:
+        "qc/{sample}/{sample}.samtools-stats.txt",
+    params:
+        extra="-t " + config["bed"]["bedfile"],
+    log:
+        "logs/qc/samtools_stats/{sample}.log",
+    singularity:
+        config["singularity"]["samtools"]
+    shell:
+        "(samtools stats {params.extra} {input} > {output} ) &> {log}"
```
Comment (Smeds) [inline src/Snakemake/rules/QC/samtools.smk]: [Wrapper "0.72.0/bio/samtools/stats"](https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/samtools/stats.html)

File: src/Snakemake/rules/QC/samtools.smk [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    input:
+        bam="DNA_bam/{sample}-ready.bam",
+    output:
+        "qc/{sample}/{sample}.samtools-stats.txt",
+    params:
+        extra="-t " + config["bed"]["bedfile"],
+    log:
+        "logs/qc/samtools_stats/{sample}.log",
+    singularity:
+        config["singularity"]["samtools"]
```
Comment (Smeds) [inline src/Snakemake/rules/QC/samtools.smk]: change to config["singularity"].get("samtools", config["singularity"].get("default", ""))?

File: src/Snakemake/rules/QC/picard.smk [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+rule picardHsMetrics:
+    input:
+        bam="DNA_bam/{sample}-ready.bam",
+        intervals=config["bed"]["intervals"],
+    output:
+        "qc/{sample}/{sample}.HsMetrics.txt",
+    log:
+        "logs/qc/picard/HsMetrics/{sample}.log",
+    singularity:
+        config["singularity"]["picard"]
```
Comment (Smeds) [inline src/Snakemake/rules/QC/picard.smk]: maybe change to 
config["singularity"].get("picard", config["singularity"].get("default", "")) and add
```
- picard=2.25.0
```
 to  https://github.com/clinical-genomics-uppsala/Twist_DNA/blob/develop/dockerfiles/somatic/environment.yml 

and update https://github.com/clinical-genomics-uppsala/Twist_DNA/blob/develop/dockerfiles/somatic/VERSION with new version (0.0.15)

File: src/Snakemake/rules/QC/picard.smk [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+rule picardInsertSize:
+    input:
+        bam="DNA_bam/{sample}-ready.bam",
+    output:
+        txt="qc/{sample}/{sample}.insert_size_metrics.txt",
+        pdf="qc/{sample}/{sample}.insert_size_histogram.pdf",
+    log:
+        "logs/qc/picard/InsertSize/{sample}.log",
+    singularity:
+        config["singularity"]["picard"]
```
Comment (Smeds) [inline src/Snakemake/rules/QC/picard.smk]: maybe change to 
config["singularity"].get("picard", config["singularity"].get("default", "")) and add
```
- picard=2.25.0
```
 to  https://github.com/clinical-genomics-uppsala/Twist_DNA/blob/develop/dockerfiles/somatic/environment.yml 

and update https://github.com/clinical-genomics-uppsala/Twist_DNA/blob/develop/dockerfiles/somatic/VERSION with new version (0.0.15)

File: src/Snakemake/rules/QC/picard.smk [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+rule PicardAlignmentSummaryMetrics:
+    input:
+        bam="DNA_bam/{sample}-ready.bam",
+        ref=config["reference"]["ref"],
+    output:
+        "qc/{sample}/{sample}.alignment_summary_metrics.txt",
+    log:
+        "logs/qc/picard/AlignmentSummaryMetrics/{sample}.log",
+    singularity:
+        config["singularity"]["picard"]
```
Comment (Smeds) [inline src/Snakemake/rules/QC/picard.smk]: maybe change to 
config["singularity"].get("picard", config["singularity"].get("default", "")) and add
```
- picard=2.25.0
```
 to  https://github.com/clinical-genomics-uppsala/Twist_DNA/blob/develop/dockerfiles/somatic/environment.yml 

and update https://github.com/clinical-genomics-uppsala/Twist_DNA/blob/develop/dockerfiles/somatic/VERSION with new version (0.0.15)

**Review by Smeds — COMMENTED** [review]
We should add a docker entry for wp3 https://github.com/clinical-genomics-uppsala/Twist_DNA/tree/develop/dockerfiles with the software need to run Wp3 pipeline

File: src/Snakemake/rules/QC/picard.smk (hunk 1 above)
Comment (jonca79) [reply src/Snakemake/rules/QC/picard.smk:16]: Wrapper needs more input files than we use at the moment.

Comment (Smeds) [reply src/Snakemake/rules/QC/picard.smk:16]: Why are we not using reference as input?

File: src/Snakemake/rules/QC/samtools.smk (hunk 5 above)
Comment (jonca79) [reply src/Snakemake/rules/QC/samtools.smk]: done

File: src/Snakemake/rules/QC/samtools.smk (hunk 6 above)
Comment (jonca79) [reply src/Snakemake/rules/QC/samtools.smk]: done

File: src/Snakemake/rules/QC/picard.smk (hunk 7 above)
Comment (jonca79) [reply src/Snakemake/rules/QC/picard.smk]: done

File: src/Snakemake/rules/QC/picard.smk (hunk 8 above)
Comment (jonca79) [reply src/Snakemake/rules/QC/picard.smk]: done

File: src/Snakemake/rules/QC/picard.smk (hunk 9 above)
Comment (jonca79) [reply src/Snakemake/rules/QC/picard.smk]: done

File: src/Snakemake/rules/QC/picard.smk (hunk 3 above)
Comment (jonca79) [reply src/Snakemake/rules/QC/picard.smk]: done

File: src/Snakemake/rules/QC/picard.smk (hunk 2 above)
Comment (jonca79) [reply src/Snakemake/rules/QC/picard.smk]: done

File: src/Snakemake/rules/QC/picard.smk (hunk 1 above)
Comment (jonca79) [reply src/Snakemake/rules/QC/picard.smk:16]: Don't know. Needs to be investigated.

Comment (jonca79) [reply src/Snakemake/rules/QC/picard.smk:16]: Submitted as an issue (enhancement).

Comment (Smeds) [reply src/Snakemake/rules/QC/picard.smk:16]: #69

**Review by Smeds — APPROVED** [review]

**Comment by jonca79** [comment]
Added a RD directory for dockers.

File: dockerfiles/RD/environment.yml [hunk 10]
```diff
@@ -0,0 +1,18 @@
+name: gms_somatic
```
Comment (Smeds) [inline dockerfiles/RD/environment.yml]: change to RD?

**Review by Smeds — APPROVED** [review]

**Pull request merged**